kv/txn_coord_sender.go

```diff
@@ -758,28 +756,26 @@ func (tc *TxnCoordSender) heartbeat(id string) {
 					Reply: reply,
 				}
 
-				task := tc.stopper.StartTask()
-				if !task.Ok() {
+				if !tc.stopper.RunTask(func() {
```

Comment:
I don't think this feels half-baked; it's a clear improvement from the previous version and is no worse than the use of `if task.OK()` from before. I would actually point to the code you linked to as *justification* for the return values; I'm unconvinced that removing the return values will make it any better.